scipy/io/_mmio.py

```diff
@@ -20,6 +22,24 @@
 
 __all__ = ['mminfo', 'mmread', 'mmwrite', 'MMFile']
 
+"""
+Maximum number of digits for `mmwrite` function.
+This can be calculated by below python codes:
+eps = np.finfo(np.float64).eps
+max_precision = np.floor(-np.log10(eps)).astype(int)
+"""
+MAX_PRECISION = 15
+
+
+def _validate_precision(precision):
+    if precision is not None:
+        # check the input can be converted to an integer
+        precision = operator.index(precision)
```

Comment:
This is an unusual way (for me at least) to test convertability to an integer. Can we do something less magical?

Reply:
This is quite a standard way of doing things across SciPy so unless there's a really nice way I'm not sure we should break the norm https://github.com/search?q=repo%3Ascipy%2Fscipy%20operator.index&type=code